### PR TITLE
Fix #163: Bump Scala.js to 0.6.5 and @js.native everything.

### LINKED ISSUE #163: Move to scala-js 0.6.5
0.6.5 adds `|`, which will enable tightening up a lot of the signatures, so I'd vote to move to it if possible.
On the other hand, if we're intentionally staying with 0.6.2 for backwards compatibility, then perhaps we shouldn't just yet.  Thoughts?

### PR DESCRIPTION
**Pull request by mseddon**
Horay, union types! :)

### CODE REVIEW

**Comment by mseddon** [comment]
Okay that failure was travis's fault, but this one char' PR is also firing warnings everywhere on the omission of `@js.native`. Now I see why the version was held back :P

I'll fix this up fully tommorrow.

**Comment by mseddon** [comment]
image

**Comment by mseddon** [comment]
That's all, hopefully, appears to work against my projects at least.

**Comment by sjrd** [comment]
> so I hoisted it for lack of a better solution.

I did not understand that.

**Comment by mseddon** [comment]
Just adding a `@js.native` annotation to that class raised the error, so I moved the definition out of the Vibration object to toplevel, which kept the compiler happy.

**Comment by sjrd** [comment]
Oh. It should be kept where it was, but make it a `trait` instead, if it's a pimp.

**Comment by mseddon** [comment]
Ah, right! nice, I'll do that, then.  Particularly since this is what I point people to who want to add to `experimental` ^^

**Comment by mseddon** [comment]
Squashed together with a less silly commit message, since the initial commit would leave us in a broken state.

**Comment by mseddon** [comment]
Hmm, shall we update the scala versions to 2.10.6 and 2.11.7 while we're at it?

**Comment by mseddon** [comment]
Just fixed a formatting issue with Vibration API.  Holding off library version change for now.
I'll tweak those too tomorrow, if you like.

**Comment by sjrd** [comment]
> Hmm, shall we update the scala versions to 2.10.6 and 2.11.7 while we're at it?

Yes, you can do that as well.

**Comment by sjrd** [comment]
> Fix #163: Bump scala-js to 0.6.5 and @js.native everything.

Please spell "Scala.js" the right way ^^

File: src/main/scala/org/scalajs/dom/experimental/Vibration.scala [hunk 1]
```diff
@@ -24,4 +25,4 @@ object Vibration {
      */
     def vibrate(pattern: js.Array[Double]): Boolean = js.native
   }
-}
+}
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/experimental/Vibration.scala]: Missing EOL at EOF.

**Comment by mseddon** [comment]
> Please spell "Scala.js" the right way ^^

*cough*. Whoops. I was thinking of the artifact :)

File: build.sbt [hunk 2]
```diff
@@ -11,9 +11,9 @@ version := "0.9.0-SNAPSHOT"
 
 organization := "org.scala-js"
 
-scalaVersion := "2.11.6"
+scalaVersion := "2.11.7"
 
-crossScalaVersions := Seq("2.10.5", "2.11.6")
+crossScalaVersions := Seq("2.10.6", "2.11.7")
```
Comment (sjrd) [inline build.sbt]: You should also update `.travis.yml`.

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
LGTM, thanks :)

**Pull request merged**